setup-elytron-guides/setup_elytron_linux.adoc

```diff
@@ -0,0 +1,148 @@
+---
+layout: setup-elytron
+title: Setting up WildFly Elytron on Linux & WSL
+permalink: /guides/setup-elytron-linux/
+os: linux
+---
+:imagesdir: ../../assets/images/setup-elytron/
+
+:toc: macro
+:toc-title:
+
+This page will guide you through the steps needed to setup a development environment for WildFly Elytron on Linux, or the https://docs.microsoft.com/en-us/windows/wsl/about[Windows Subsystem for Linux (WSL)]. For WSL developers, all commands should be entered into the WSL terminal _(i.e. not Command Prompt, PowerShell, or Git Bash)_.
+
+toc::[]
+
+== Installing Java
+
+. First, let's check if you already have Java installed.
+    .. Open a terminal, and run `java --version`.
+    .. If the output shows a version number, check to make sure it is Java 11 (e.g., "11.0.15.1"). If so, Java 11 is already installed and you can jump to the *Installing Maven* section. If the output is not Java 11, or if the output is `command not found: java` or a similar message indicating you do not have Java installed, then continue with the following steps.
+. It is easiest to install Java on Linux using the preferred package manager of your Linux distribution. Here are some examples of how to install Java on common Linux distributions:
+    ** Ubuntu: `sudo apt-get install openjdk-11-jdk`
+    ** Fedora: `sudo dnf install java-11-openjdk`
+    ** Arch: `sudo pacman -S java-11-openjdk`
+. Next, configure the default JVM version for your system:
+    ** Ubuntu: `sudo update-alternatives --config java`
+    ** Fedora: `sudo alternatives --config java`
+    ** Arch: `archlinux-java set java-11-openjdk`
+. Configure JAVA_HOME and PATH environment variables for Java:
+    .. Open the configuration file for your shell, which you can identify by typing `echo $0`. If the response is `bash`, open the file _~/.bashrc_, and if it is `sh`, edit _~/.profile_. You can look up config files for other shells by searching the name of the shell, followed by "config file" (ex. "zsh config file").
+    .. Go to the end of the config file, and add the following lines:
+    +
+    [source,bash]
+    ----
+    export JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:bin/java::")
+    export PATH=$JAVA_HOME/bin:$PATH
+    ----
+
+    .. Run `source <config file>`, substituting in your config file, to load the changes into the terminal.
+    .. You can undo these changes later on by following the steps in the *Temporary Configuration* section.
```

Comment:
I probably should rename that section to **Restoring Configuration**, since that's really what it does.

The real "temporary" section is [Session Configuration Only](http://localhost:4000/wildfly-elytron/guides/setup-elytron-linux/#advanced-session-configuration-only), but that'll probably cause a lot of trouble for first-timers, so I left it as an appendix.

Reply:
I'm thinking we could leave out the Restoring Configuration section (since it makes the guide longer).

Then, for setting environment variables, we can just show both the persistent approach and the session config only approach so people can choose what's best for them.

Reply:
I kind of like the `Restoring configuraiton` section, so I would use it since we have it written. But I agree that shorter guides seem less demanding. Can we for example put Restoring config section in another page and only add a link at the end of linux guide that points them to the Restoring configuration page if they want to do that? 

Btw if we care about making the linux guide shorter, one way could be to remove the archlinux mentions. Since arch is a rolling release that requires a lot of configuration to install it, users that use it should be able to find docs on how to configure these things themselves.

About the` session configuration` - it is a pickle :D It might look better to mention it together but I agree it might cause problems for first timers given that each new terminal will not work for them as well as IDE problems. So I think it has advantages to only mention it at the end, as it is now. But I am fine with both approaches.

Reply:
+1, if we could link to a separate page for Restoring Configuration and only mention it at the end, that's fine.

I think it's ok to leave the archlinux mentions.

For session configuration, perhaps that could also be on the page with the Restoring Configuration?